lib/segment/src/id_tracker/immutable_id_tracker.rs

```diff
@@ -0,0 +1,446 @@
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::io::{BufReader, BufWriter};
+use std::mem::size_of_val;
+use std::path::{Path, PathBuf};
+
+use bitvec::prelude::BitSlice;
+use common::types::PointOffsetType;
+use memory::mmap_ops::{create_and_ensure_length, open_write_mmap};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::common::mmap_type::{MmapBitSlice, MmapSlice};
+use crate::common::operation_error::{OperationError, OperationResult};
+use crate::common::Flusher;
+use crate::id_tracker::simple_id_tracker::StoredPointId;
+use crate::id_tracker::{IdTracker, IdTrackerEnum};
+use crate::types::{PointIdType, SeqNumberType};
+
+pub const DELETED_FILE_NAME: &str = "id_tracker.deleted";
```

Comment:
Since we create new files, it should be used in `take_snapshot` of segment.

I propose to do similar to other components and put `fn files(&self) -> Vec<PathBuf>` into IdTracker trait